homeassistant/components/reolink/update.py

```diff
@@ -22,40 +23,61 @@
 
 from . import ReolinkData
 from .const import DOMAIN
-from .entity import ReolinkBaseCoordinatorEntity
-
-LOGGER = logging.getLogger(__name__)
+from .entity import ReolinkHostCoordinatorEntity, ReolinkHostEntityDescription
 
 POLL_AFTER_INSTALL = 120
 
 
+@dataclass(frozen=True, kw_only=True)
+class ReolinkHostUpdateEntityDescription(
+    UpdateEntityDescription,
+    ReolinkHostEntityDescription,
+):
+    """A class that describes host update entities."""
+
+
+HOST_UPDATE_ENTITIES = (
+    ReolinkHostUpdateEntityDescription(
+        key="firmware",
+        supported=lambda api: api.supported(None, "firmware"),
+    ),
+)
+
+
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up update entities for Reolink component."""
     reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]
-    async_add_entities([ReolinkUpdateEntity(reolink_data)])
 
+    entities: list[ReolinkHostUpdateEntity] = [
+        ReolinkHostUpdateEntity(reolink_data, entity_description)
+        for entity_description in HOST_UPDATE_ENTITIES
+        if entity_description.supported(reolink_data.host.api)
+    ]
+    async_add_entities(entities)
 
-class ReolinkUpdateEntity(
-    ReolinkBaseCoordinatorEntity[str | Literal[False] | NewSoftwareVersion],
+
+class ReolinkHostUpdateEntity(
+    ReolinkHostCoordinatorEntity,
     UpdateEntity,
 ):
-    """Update entity for a Netgear device."""
+    """Update entity class for Reolink Host."""
 
+    entity_description: ReolinkHostUpdateEntityDescription
     _attr_device_class = UpdateDeviceClass.FIRMWARE
     _attr_release_url = "https://reolink.com/download-center/"
```

Comment:
Want to move these to the entity description? 

And should we set has_entity_name=True ?

Reply:
I have moved the device_class to the enitity description, thank you for the suggestion!
But the release_url is not part of the entity description, I don't think it is needed to add it in the mixin dataclass.

has_entity_name is already set to True in the inherited class `ReolinkHostCoordinatorEntity`.
I think it should also be True.